server/src-lib/Hasura/GraphQL/Execute/Insert.hs

```diff
@@ -66,23 +65,23 @@ traverseAnnInsert f (AnnInsert fieldName isSingle (annIns, mutationOutput)) =
           )
       <*> pure columns
       <*> traverse f defaultValues
-    traverseObject (AnnInsObj columns objRels arrRels) = AnnInsObj
+    traverseObject (IR.AnnInsObj columns objRels arrRels) = IR.AnnInsObj
       <$> traverse (traverse f) columns
       <*> traverse (traverseRel traverseSingle) objRels
       <*> traverse (traverseRel traverseMulti)  arrRels
-    traverseRel z (RelIns object relInfo) = RelIns <$> z object <*> pure relInfo
+    traverseRel z (IR.RelIns object relInfo) = IR.RelIns <$> z object <*> pure relInfo
 
 
 convertToSQLTransaction
   :: (HasVersion, MonadTx m, MonadIO m, Tracing.MonadTrace m)
   => Env.Environment
-  -> AnnInsert 'Postgres S.SQLExp
+  -> IR.AnnInsert 'Postgres PG.SQLExp
   -> PGE.MutationRemoteJoinCtx
   -> Seq.Seq Q.PrepArg
   -> Bool
   -> m EncJSON
-convertToSQLTransaction env (AnnInsert fieldName isSingle (annIns, mutationOutput)) remoteJoinCtx planVars stringifyNum =
-  if null $ _aiInsObj annIns
+convertToSQLTransaction env (IR.AnnInsert fieldName isSingle (annIns, mutationOutput)) remoteJoinCtx planVars stringifyNum =
```

Comment:
This probably needs to go into the `Postgres` specific module?

Reply:
Yup; but so far I haven't started splitting `GraphQL.Execute`.